superset/charts/api.py

```diff
@@ -503,6 +504,50 @@ def data(self) -> Response:
 
         return response
 
+    @expose("/data/stop", methods=["POST"])
+    @event_logger.log_this
+    @protect()
+    @safe
+    @statsd_metrics
+    def data_stop(self) -> Response:
+        """
+        Takes a dashboard id and tries to cancel all associated chart data requests
+        issued by the user.
+        ---
+        post:
+          description: >-
+            Takes a dashboard id and tries to cancel all associated chart data requests
+            issued by the user
+          requestBody:
+            description: >-
+              The dashboard id.
+            required: true
+            content:
+              application/json:
+                schema:
+                  $ref: "#/components/schemas/ChartDataStopSchema"
+          responses:
+            200:
+              description: Pending dashboard queries terminated
+              content:
+                application/json:
+                  schema:
+                    type: object
+            400:
+              $ref: '#/components/responses/400'
+            500:
+              $ref: '#/components/responses/500'
+        """
+        if request.is_json:
+            json_body = request.json
+            dashboard_id = json_body.get("dashboard_id")
+            if not dashboard_id:
+                return self.response(400, message="dashboard_id missing in body")
+            hook = current_app.config["STOP_DASHBOARD_PENDING_QUERIES_HOOK"]
+            hook(dashboard_id, g.user.username)
```

Comment:
`@protect` or `@has_access` does not guarantee we a have a user, since it's possible for a public/non authenticated user to access this resource. It's probably very unlikely that a `Public` role has access to this resource, but now it's possible to set `PUBLIC_ROLE_LIKE = "Admin"`.